src/test/java/com/alkemy/ong/integration/organization/OrganizationPublicEndpointIntegrationTest.java

```diff
@@ -0,0 +1,37 @@
+package com.alkemy.ong.integration.organization;
+
+import static org.junit.Assert.assertEquals;
+
+import com.alkemy.ong.common.AbstractBaseIntegrationTest;
+import com.alkemy.ong.service.OrganizationServiceImpl;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class OrganizationPublicEndpointIntegrationTest extends AbstractBaseIntegrationTest {
+
+  private final String PATH = "/organization/public";
+
+  @MockBean
+  OrganizationServiceImpl organizationService;
```

Comment:
Si mockeas el organizationService, esto ya no sería un integration test, sino que sería un unit test. Básicamente lo que hace este código es cuando llama a organizationService.getOrganizationWithSlides() returna null y como eso no es un problema para el controller, el controller devuelve OK y tenemos un green test. Sin embargo, no esta testeando nada ni verificando nada que sea util en caso de que se hagan modificaciones al código en una posterior iteración.

Si esto fuera un unit test de la clase controller, mockearia el servicio, ya que lo único que me importa en el contexto del controller es que devuelva un 200 y que se llame al organizationService. FOR YOUR INFORMATION: Para verificar que efectivamente haya iteración con el service puedo usar el metodo verify de Mockito.

Pero esto no es un unit test, es un test de integración... Por lo tanto, el código debe pasar por el service real, no debo mockearlo. Lo único mockeado en este test deberían ser los repositorios. Al mockear los repositorios, vas a tener acceso a poder verificar la response del servicio con algo como:

```
    OrganizationResponse responseBody = response.getBody();
    assertNotNull(responseBody);
    assertEquals(responseBody.getEmail(), "whatever is set in the stub class");
    assertEquals(responseBody.getImage(), "whatever is set in the stub class");
```

Reply:
Adicionalmente, los test de integración en un proyecto real no suelen contener mocks. Salvo que sean isolation test. Para minimizar el scope de la tarea estamos utilizando mocks en los repositorios para los test de integración y de paso les permite a ustedes comprender el uso de un mock. Ya que los mismo son la base para los unit test.